server/controllers/packFeed.js

```diff
@@ -0,0 +1,90 @@
+const db = require('../db/database');
+var getUserPacksId = (userId, res) => {
+  // db.connect();
+  //replace below query with: `select * from public."packUsers" where public."packUsers"."userId" = ${userId} or public."packUsers"."userId" = 15`
+  //to test if it works with multiple groups being returned by query.
+  db.query(`select * from pack_users where pack_users.user_id = ${userId} `).then((partOf) => {
+    var promises = [];
+    for (var i = 0; i < partOf.rows.length; i++) {
+      promises.push(
+        db.query(`select * from packs where packs.pack_id = ${partOf.rows[i].pack_id}`)
+      );
+    }
+    var toSendBack = [];
+    Promise.all(promises).then((result) => {
+      for (var i = 0; i < result.length; i++) {
+        for (var j = 0; j < result[i].rows.length; j++) {
+          toSendBack.push(result[i].rows[j]);
+        }
+      }
+      // console.log('sending back', toSendBack);
+      res.status(201).send(toSendBack);
+    });
+  });
+};
+
+var getUserInformation = (userId) => {
+  db.query(`select * from users where users.user_id = ${userId}`)
+    .then((res) => {
+      // console.log('Query Response', res.rows);
+      return res.rows;
+    })
+    .catch((err) => {
+      console.log('Err occured', err);
+    });
+};
+
+var getPackPosts = (packId) => {
+  return db.query(`select * from posts where posts.pack_id = ${packId}`);
+};
+
+var getAllPostsFromAllPacks = (userId, resServer) => {
```

Comment:
Not sure if this is faster, but if helpful, you could try this query:

SELECT * FROM 
	( SELECT pack_id, name FROM packs WHERE pack_id IN (SELECT pack_id FROM pack_users WHERE user_id = 7) ) a
	LEFT JOIN 
	( SELECT pack_id, array_agg(json_build_object('id', post_id, 'user_id', user_id, 'body', body, 'date', date, 'photo_url', photo_url )) FROM posts GROUP BY pack_id ) b
	ON a.pack_id = b.pack_id
	
which will return pack_id, pack_name, and an array of post objects, each with body, user, date, photo, and id properties 

No need to adjust if this is more trouble than it's worth - mentioning in case helpful for this or future queries!